### ISSUE TITLE
Need clarity on how the core and plugins interact over annotation lifecycles

### ISSUE THREAD

**Issue by greg-pendlebury** [openannotation/annotator#499]
As per discussion: http://permalink.gmane.org/gmane.comp.web.annotator/1379

It would be good to have some solid information in the doco which can inform code design, both inside the core and for plugin authors on the best ways to integrate. Things like:
 * What methods are considered APIs?
 * What is the normal way of doing things like replacing 'add' triggers, editors and content display elements?
 * Similar to above, but what are the implications of doing any of those things if they disrupt lifecycle events?

**Comment by greg-pendlebury** [comment]
A good example for addition. The core library fires 'beforeAnnotationCreated' and 'annotationCreated'. The Store plugin listens to the second event and stores the created annotation, typically supplementing it with additional data generated by the server during storage.

However the Store plugin directly drops the new annotation data into the core data structures without firing any events. There is no way for other interested plugins to interact with this other than brute force polling the core data structures.. We locally modify our Store plugin to then fire an 'afterAnnotationStored' event

**Comment by chrismichaels84** [comment]
+1

**Comment by tilgovi** [comment]
For v2 we have this in the docs: http://docs.annotatorjs.org/en/latest/module-development.html#module-hooks

Feedback welcome! If anyone wants to contribute to documentation for v1.2.x please make a pull request and we'd be glad to merge it.

**Comment by tilgovi** [comment]
Going to close this. Please re-open or post a new issue with any confusion about the v2.0 documentation.

**Comment by greg-pendlebury** [comment]
Personally, I wouldn't close this ticket, but won't cry if you want to. The v2 doco (specifically in relation to module hooks) is pretty similar to v1 events, so I can't see a lot of new material there. Things like the lifecycle event I made reference to ('afterAnnotationStored') which we had to create ourselves are highlighting the gaps I was trying to find in the core library's interaction with plugins.
